state/protocol/protocol_state/mutator_test.go

```diff
@@ -0,0 +1,72 @@
+package protocol_state
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	storerr "github.com/onflow/flow-go/storage"
+	"github.com/onflow/flow-go/storage/badger/transaction"
+	storagemock "github.com/onflow/flow-go/storage/mock"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+func TestProtocolStateMutator(t *testing.T) {
+	suite.Run(t, new(MutatorSuite))
+}
+
+type MutatorSuite struct {
+	suite.Suite
+	protocolStateDB *storagemock.ProtocolState
+
+	mutator *Mutator
+}
+
+func (s *MutatorSuite) SetupTest() {
+	s.protocolStateDB = storagemock.NewProtocolState(s.T())
+	s.mutator = NewMutator(s.protocolStateDB)
+}
+
+// TestCreateUpdaterForUnknownBlock tests that CreateUpdater returns an error if the parent protocol state is not found.
+func (s *MutatorSuite) TestCreateUpdaterForUnknownBlock() {
+	candidate := unittest.BlockHeaderFixture()
+	s.protocolStateDB.On("ByBlockID", candidate.ParentID).Return(nil, storerr.ErrNotFound)
+	updater, err := s.mutator.CreateUpdater(candidate)
+	require.ErrorIs(s.T(), err, storerr.ErrNotFound)
```

Comment:
We should document this as an expected error, or wrap as `irrecoverable`.

Reply:
I think from the perspective of the `Mutator` the error should be benign, as it allows (under certain conditions, e.g. access node trying to serve an API request) to continue. 

See [my comment below](https://github.com/onflow/flow-go/pull/4597#issuecomment-1709526474) for a detailed suggestion on documenting error returns of [interface `ProtocolState`](https://github.com/onflow/flow-go/blob/ddd1a1c485ad32e7547b604431d7d9ed696c1026/storage/protocol_state.go).

Reply:
⚠️  Approving the PR - but please address this before merging.

Reply:
I've updated godoc so we expect `ErrNotFound` [`a1969a0` (#4597)](https://github.com/onflow/flow-go/pull/4597/commits/a1969a07843fd155d33e7f89a3c9e65136293261)